daffodil-runtime1/src/main/scala/org/apache/daffodil/processors/DataProcessor.scala

```diff
@@ -243,6 +250,18 @@ class DataProcessor private (
 
   def withValidationMode(mode:ValidationMode.Type): DataProcessor = copy(validationMode = mode)
 
+  def withValidator(validator: Validator): DataProcessor = copy(validationMode = ValidationMode.Custom(validator))
+
+  lazy val validator: Validator = {
+    validationMode match {
+      case ValidationMode.Custom(cv) => cv
+      case _ =>
+        val v = ConfigValueFactory.fromAnyRef(ssrd.elementRuntimeData.schemaURIStringsForFullValidation)
+        val f = Validators.default
+        f.make(ConfigFactory.parseMap(Map(f.name() -> v).asJava))
+    }
```

Comment:
This is a bit interseting. So Custom Validators won't have any access to the schemaURIStrings? Is it possible other custom validators will need those? I imaginge in something like schematron it might not, but other XML Schema validators that aren't our built-in xerces one might? Seems odd that only our internal validator gets access to this information? Or shold no Validators get this access, including our own, and Validators are expected to implement their own logic for this like URI resolution?

Reply:
Handling passing this to Xerces as a special case right now because it happens to be the "default" fallback validator here and it happens to need this bit of config.

Any other validator would be built outside this scope and should have already processed its configuration.

So in my mind its a special case that catches the eye, but no change is needed.

I did refactor this a bit to make it more explicit that this is a requirement of the Xerces validator, not a requirement of being the default.